src/libsystemd/sd-id128/id128-util.c

```diff
@@ -152,14 +158,17 @@ int id128_write_fd(int fd, Id128FormatFlag f, sd_id128_t id) {
         return 0;
 }
 
-int id128_write(const char *p, Id128FormatFlag f, sd_id128_t id) {
-        _cleanup_close_ int fd = -EBADF;
+int id128_write(const char *root, const char *path, Id128FormatFlag f, sd_id128_t id) {
+        _cleanup_free_ char *filename = NULL;
+        _cleanup_close_ int dir_fd = -EBADF;
 
-        fd = open(p, O_WRONLY|O_CREAT|O_CLOEXEC|O_NOCTTY|O_TRUNC|(FLAGS_SET(f, ID128_NOFOLLOW) ? O_NOFOLLOW : 0), 0444);
-        if (fd < 0)
-                return -errno;
+        assert(path);
+
+        dir_fd = chase_and_open_parent(path, root, CHASE_PREFIX_ROOT | (FLAGS_SET(f, ID128_NOFOLLOW) ? CHASE_NOFOLLOW : 0), &filename);
+        if (dir_fd < 0)
+                return dir_fd;
 
-        return id128_write_fd(fd, f, id);
+        return id128_write_at(dir_fd, filename, f, id);
```

Comment:
So regardless of whether the user sets ID128_NOFOLLOW when calling id128_write(), we've already resolved all symlinks we want to resolve at this point, so we should always add ID128_NOFOLLOW when calling id128_write_at() here so we definitely don't resolve any symlinks anymore.